beamer/tests/config/test_write.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import tempfile
+from pathlib import Path
+from typing import Any
+
+import ape
+import pytest
+from eth_utils import to_checksum_address
+
+import beamer.config.commands
+from beamer.config.state import ChainConfig
+from beamer.deploy.artifacts import Deployment
+from beamer.tests.config.util import (
+    CommandFailed,
+    deploy,
+    read_config_state,
+    run,
+    write_keystore_file,
+)
+
+
+def _write_config_state(rpc_file, artifact, account, current_config, desired_config):
+    with tempfile.TemporaryDirectory() as tmp_path:
+        current_state_path = Path(tmp_path) / "current.state"
+        current_config.to_file(current_state_path)
+
+        desired_state_path = Path(tmp_path) / "desired.state"
+        desired_config.to_file(desired_state_path)
+
+        keystore_file = Path(tmp_path) / "key.json"
+        write_keystore_file(keystore_file, account.private_key, "test")
+
+        root = Path(__file__).parents[3]
+        run(
+            beamer.config.commands.write,
+            (
+                "--rpc-file",
+                rpc_file,
+                "--abi-dir",
+                f"{root}/contracts/.build/",
+                "--artifact",
+                artifact,
+                "--keystore-file",
+                str(keystore_file),
+                "--password",
+                "test",
+                str(current_state_path),
+                str(desired_state_path),
+            ),
+        )
+
+
+@pytest.fixture
+def tmp_deployment_path(tmp_path_factory):
+    return tmp_path_factory.mktemp("deployment")
+
+
+@pytest.fixture
+def deployment(tmp_deployment_path, deployer):
```

Comment:
maybe we can change the name to deployment_summary just to differentiate between deployment instance below

Reply:
Decided on `deployment_objects` in a joint discussion.